File: ios/event/BluetoothDeviceEvent.swift
//
//  BluetoothDeviceEvent.swift
//  react-native-bluetooth-classic
//
//  Wraps device event data within an event containing the device and timestamp.
//  This is used for device connection/disconnection/discovery events.
//

import Foundation

class BluetoothDeviceEvent: NSObject, Mappable {
    public private(set) var device: NativeDevice
    public private(set) var timestamp: Date
    public private(set) var eventType: EventType
    
    init(eventType: EventType, device: NativeDevice) {
        self.eventType = eventType
        self.device = device
        self.timestamp = Date()
    }
    
    func map() -> NSDictionary {
        return [
            "eventType": eventType.name,
            "device": device.map(),
            "timestamp": ISO8601DateFormatter().string(from: timestamp)
        ]
    }
}

File: android/src/main/java/kjd/reactnative/bluetooth/event/BluetoothDeviceEvent.java
package kjd.reactnative.bluetooth.event;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.WritableMap;

import kjd.reactnative.bluetooth.device.NativeDevice;

/**
 * Wraps device event data within an event containing the device and timestamp.
 * This is used for device connection/disconnection/discovery events.
 *
 * @author kendavidson
 */
public class BluetoothDeviceEvent extends BluetoothEvent {

    private NativeDevice device;

    public BluetoothDeviceEvent(EventType eventType, NativeDevice device) {
        super(eventType);
        this.device = device;
    }

    @Override
    public ReadableMap buildMap() {
        WritableMap map = Arguments.createMap();
        map.putMap("device", device.map());
        return map;
    }
}
